File: Tyche.Android/bet/src/main/java/com/felipearpa/bet/domain/PoolGamblerBetRemoteDataSource.kt
package com.felipearpa.bet.domain

import com.felipearpa.core.paging.CursorPage
import retrofit2.http.Body
import retrofit2.http.GET
import retrofit2.http.PATCH
import retrofit2.http.Path
import retrofit2.http.Query

interface PoolGamblerBetRemoteDataSource {

    @GET("/pool/{poolId}/gambler/{gamblerId}/bets")
    suspend fun getPoolGamblerBets(
        @Path("poolId") poolId: String,
        @Path("gamblerId") gamblerId: String,
        @Query("next") next: String? = null,
        @Query("searchText") searchText: String? = null
    ): CursorPage<PoolGamblerBetResponse>

    @PATCH("bet")
    suspend fun bet(@Body betRequest: BetRequest): PoolGamblerBetResponse
}

File: Tyche.iOS/Bet/Sources/Bet/Domain/PoolGamblerBetRemoteDataSource.swift
import Core

protocol PoolGamblerBetRemoteDataSource {
    func getPoolGamblerBets(
        poolId: String,
        gamblerId: String,
        next: String?,
        searchText: String?
    ) async throws -> CursorPage<PoolGamblerBetResponse>
    
    func bet(betRequest: BetRequest) async throws -> PoolGamblerBetResponse
}
